// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package mypackage;

// The greeting service definition.
service MyService {
  rpc GetSqrt (SqrtRequest) returns (SqrtReply) {}
  rpc GetSigma (stream SigmaRequest) returns (SigmaReply) {}
  rpc GetMax (stream MaxRequest) returns (stream MaxReply) {}
  //rpc SayHello (HelloRequest) returns (HelloReply) {}
  //rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
}

message SqrtRequest {
  float val = 1;
}

message SqrtReply {
  float res = 1;
}

message SigmaRequest {
  float val = 1;
}

message SigmaReply {
  float res = 1;
}

message MaxRequest {
  float val = 1;
}

message MaxReply {
  float res = 1;
}
